### PR TITLE
Moved all streams into a try with resources setup

### PR DESCRIPTION
**Pull request by azoitl**

### CODE REVIEW

File: org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java [hunk 1]
```diff
@@ -90,8 +90,12 @@ public abstract class Shape extends ModelElement {
 	 * @see #setPropertyValue(Object, Object)
 	 */
 	static {
-		descriptors = new IPropertyDescriptor[] { new TextPropertyDescriptor(XPOS_PROP, ShapesExampleMessages.Shape_X), // id and description pair
-				new TextPropertyDescriptor(YPOS_PROP, ShapesExampleMessages.Shape_Y), new TextPropertyDescriptor(WIDTH_PROP, ShapesExampleMessages.Shape_Width),
+		descriptors = new IPropertyDescriptor[] { new TextPropertyDescriptor(XPOS_PROP, ShapesExampleMessages.Shape_X), // id
+																														// and
+																														// description
+																														// pair
+				new TextPropertyDescriptor(YPOS_PROP, ShapesExampleMessages.Shape_Y),
```
Comment (ptziegler) [inline org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java:97]: Just to fix the broken comment:
```suggestion
		// id and description pair
		descriptors = new IPropertyDescriptor[] { new TextPropertyDescriptor(XPOS_PROP, ShapesExampleMessages.Shape_X),
				new TextPropertyDescriptor(YPOS_PROP, ShapesExampleMessages.Shape_Y),
				new TextPropertyDescriptor(WIDTH_PROP, ShapesExampleMessages.Shape_Width),
				new TextPropertyDescriptor(HEIGHT_PROP, ShapesExampleMessages.Shape_Height), };
```

File: org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java [hunk 2]
```diff
@@ -108,13 +112,12 @@ public abstract class Shape extends ModelElement {
...
-		Image image = new Image(null, stream);
-		try {
-			stream.close();
+		try (InputStream stream = ShapesPlugin.class.getResourceAsStream(name)) {
+			return new Image(null, stream);
 		} catch (IOException ioe) {
+			ioe.printStackTrace();
 		}
-		return image;
+		return null;
```
Comment (ptziegler) [inline org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java:120]: Same as in TestImages.

File: org.eclipse.gef.examples.logic/src/org/eclipse/gef/examples/logicdesigner/model/LogicSubpart.java [hunk 3]
```diff
@@ -51,13 +51,12 @@ public abstract class LogicSubpart extends LogicElement {
...
-		Image image = new Image(null, stream);
-		try {
-			stream.close();
+		try (InputStream stream = rsrcClass.getResourceAsStream(name)) {
+			return new Image(null, stream);
 		} catch (IOException ioe) {
+			ioe.printStackTrace();
 		}
-		return image;
+		return null;
```
Comment (ptziegler) [inline org.eclipse.gef.examples.logic/src/org/eclipse/gef/examples/logicdesigner/model/LogicSubpart.java:59]: Same as in TestImages.

File: org.eclipse.draw2d.tests/src/org/eclipse/draw2d/test/TestImages.java [hunk 4]
```diff
@@ -19,23 +19,16 @@
...
+		try (InputStream is = TestImages.class.getResourceAsStream(imageName)) { // $NON-NLS-1$
+			return new Image(null, is);
 
-		is = TestImages.class.getResourceAsStream("icons/bits24.jpg"); //$NON-NLS-1$
-		depth_24 = new Image(null, is);
-		try {
-			is.close();
 		} catch (Exception e) {
 		}
+		return null;
```
Comment (ptziegler) [inline org.eclipse.draw2d.tests/src/org/eclipse/draw2d/test/TestImages.java:31]: This whole expression I would simplify to `ImageDescriptor.createFromFile(rsrcClass, name).createImage()` / `ImageDescriptor.createFromFile(TestImages.class, name).createImage()`.

Comment (azoitl) [reply org.eclipse.draw2d.tests/src/org/eclipse/draw2d/test/TestImages.java]: Thx for pointing this out. Changed!

File: org.eclipse.gef.examples.logic/src/org/eclipse/gef/examples/logicdesigner/model/LogicSubpart.java (hunk 3 above)
Comment (azoitl) [reply org.eclipse.gef.examples.logic/src/org/eclipse/gef/examples/logicdesigner/model/LogicSubpart.java]: Thx for pointing this out. Changed!

File: org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java (hunk 2 above)
Comment (azoitl) [reply org.eclipse.gef.examples.shapes/src/org/eclipse/gef/examples/shapes/model/Shape.java]: Thx for pointing this out. Changed!

**Review by ptziegler — APPROVED** [review]

**Pull request merged**